scripts/automation/tests/run.py

```diff
@@ -23,29 +51,35 @@ def run_tests(modules, parallel, run_live):
     if run_live:
         os.environ['AZURE_TEST_RUN_LIVE'] = 'True'
 
-    # get test runner
-    run_nose = get_nose_runner(test_results_folder, xunit_report=False, parallel=parallel,
-                               process_timeout=3600 if run_live else 600)
+    if not tests:
+        # the --test is not given, use nosetests to run entire module
+        print('Drive test by nosetests')
+        runner = get_nose_runner(test_results_folder, parallel=parallel, process_timeout=3600 if run_live else 600)
+    else:
+        # the --test is given, use unittest to run single test
+        print('Drive test by unittest')
+        runner = get_unittest_runner(tests)
 
     # run tests
-    result, test_report = run_nose([p for _, _, p in modules])
-
-    print('==== TEST RESULT ====\n{}'.format(test_report))
+    result = runner([p for _, _, p in modules])
 
     return result
 
 
 if __name__ == '__main__':
     parse = argparse.ArgumentParser('Test tools')
     parse.add_argument('--module', dest='modules', action='append',
-                       help='The modules of which the test to be run. Accept short names, except '
-                            'azure-cli, azure-cli-core and azure-cli-nspkg. The modules list can '
-                            'also be set through environment variable AZURE_CLI_TEST_MODULES. The '
-                            'value should be a string of comma separated module names. The '
-                            'environment variable will be overwritten by command line parameters.')
+                       help='The modules of which the test to be run. Accept short names, except azure-cli, '
+                            'azure-cli-core and azure-cli-nspkg. The modules list can also be set through environment '
+                            'variable AZURE_CLI_TEST_MODULES. The value should be a string of comma separated module '
```

Comment:
A little strange--never knew this. Why not make it space separated using the native argparse supported nargs='+'?